File: Solution/1_EASY/1_1_TwoNumber.js
//Time Complexity O(n^2) Space O(n)
function twoNumberSum(array, targetSum) {
  for (let i = 0; i < array.length - 1; i++) {
    let firstNum = array[i];
    for (let j = i + 1; j < array.length; j++) {
      let secondSum = array[j];
      if (firstNum + secondSum === targetSum)
        return [firstNum, secondSum].sort((a, b) => a - b);
    }
  }
  return [];
}

console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 10));
console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 11));
console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 99));

File: Solution/1_EASY/1_1_TwoNumber.cpp
#include <vector>
#include <iostream>
using namespace std;
vector<int> twoNumberSum(vector<int> array, int targetSum)
{
    for (int i = 0; i < array.size() - 1; i++)
    {
        int firstNum = array[i];
        for (int j = i+1; j < array.size(); j++)
        {
            int secondNum =array[j];
            if (array[i] + array[j] == targetSum)
            {
                return firstNum > secondNum ? vector<int> (secondNum,firstNum):vector<int> (firstNum,secondNum);
            }
        }
    }
    return {};
}

int main()
{
    vector<int> array =[3, 5, -4, 8, 11, 1, -1, 6];
    int target =10;

    cout << twoNumberSum(array,target);
    return 0;
}